### PR TITLE
roachtest: initialize system before restricting lease transfers

### LINKED ISSUE #110842: roachtest: failover/partial/lease-leader/lease=expiration failed
roachtest.failover/partial/lease-leader/lease=expiration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/11814636?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/11814636?buildTab=artifacts#/failover/partial/lease-leader/lease=expiration) on master @ [6cbd07ee6fbfb92706e8cdc8c559960b1bc41663](https://github.com/cockroachdb/cockroach/commits/6cbd07ee6fbfb92706e8cdc8c559960b1bc41663):

```
(test_runner.go:1141).runTest: test timed out (30m0s)
(assertions.go:333).Fail: 
	Error Trace:	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/failover.go:1674
	            				github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/failover.go:568
	            				main/pkg/cmd/roachtest/test_runner.go:1116
	            				src/runtime/asm_amd64.s:1598
	Error:      	Received unexpected error:
	            	dial tcp 34.74.6.7:26257: connect: connection refused
	Test:       	failover/partial/lease-leader/lease=expiration
(require.go:1360).NoError: FailNow called
(cluster.go:2141).Run: context canceled
test artifacts and logs in: /artifacts/failover/partial/lease-leader/lease=expiration/run_1
```
<p>Parameters: <code>ROACHTEST_arch=amd64</code>
, <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=2</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>

See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)

See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-11814636/failover-partial-lease-leader-lease-expiration/1695055822976/1695057777290)

</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*failover/partial/lease-leader/lease=expiration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

Jira issue: CRDB-31644

### LINKED ISSUE #111068: roachtest: failover/partial/lease-leader failed
roachtest.failover/partial/lease-leader [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/11864918?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/11864918?buildTab=artifacts#/failover/partial/lease-leader) on master @ [49eef749235f15d57153cb00f48dfd3b27be6c54](https://github.com/cockroachdb/cockroach/commits/49eef749235f15d57153cb00f48dfd3b27be6c54):

```
(test_runner.go:1141).runTest: test timed out (30m0s)
(assertions.go:333).Fail: 
	Error Trace:	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/failover.go:1674
	            				github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/failover.go:568
	            				main/pkg/cmd/roachtest/test_runner.go:1116
	            				src/runtime/asm_amd64.s:1598
	Error:      	Received unexpected error:
	            	read tcp 172.17.0.3:41708->104.196.70.36:26257: read: connection reset by peer
	Test:       	failover/partial/lease-leader
(require.go:1360).NoError: FailNow called
(cluster.go:2141).Run: context canceled
test artifacts and logs in: /artifacts/failover/partial/lease-leader/run_1
```
<p>Parameters: <code>ROACHTEST_arch=amd64</code>
, <code>ROACHTEST_cloud=gce</code>
, <code>ROACHTEST_cpu=2</code>
, <code>ROACHTEST_encrypted=false</code>
, <code>ROACHTEST_ssd=0</code>
</p>
<details><summary>Help</summary>
<p>

See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)

See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-11864918/failover-partial-lease-leader/1695316103601/1695318060427)

</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*failover/partial/lease-leader.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

Jira issue: CRDB-31749

### PR DESCRIPTION
**Pull request by andrewbaptist**
The test would previously set the parameter
`COCKROACH_DISABLE_LEADER_FOLLOWS_LEASEHOLDER` prior to initializing the
kv workload wth 1K splits. This would occasionally fail due to
rebalancing of ranges post-split. This commit allows all the splits and
rebalances to happen prior to setting this flat.

Epic: none
Fixes: https://github.com/cockroachdb/cockroach/issues/111068
Fixes: https://github.com/cockroachdb/cockroach/issues/110842

Release note: None

### CODE REVIEW

File: pkg/cmd/roachtest/tests/failover.go [hunk 1]
```diff
@@ -558,18 +556,12 @@ func runFailoverPartialLeaseLeader(ctx context.Context, t test.Test, c cluster.C
 	// initial RF of 5, so pass in exactlyReplicationFactor below.
 	require.NoError(t, WaitForReplication(ctx, t, conn, 3, exactlyReplicationFactor))
 
-	// Disable the replicate queue. It can otherwise end up with stuck
-	// overreplicated ranges during rebalancing, because downreplication requires
-	// the Raft leader to be colocated with the leaseholder.
-	_, err := conn.ExecContext(ctx, `SET CLUSTER SETTING kv.replicate_queue.enabled = false`)
```
Comment (erikgrinaker) [inline pkg/cmd/roachtest/tests/failover.go:564]: Is it ok to not disable this any more? We're going to get stuck if it tries to rebalance something after the restart, no?

**Review by andrewbaptist — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/112123)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @DarrylWong, @erikgrinaker, and @herkolategan)

___
*[`pkg/cmd/roachtest/tests/failover.go` line 0 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/112123#gh-1354580056:-NgU5yDs7cO3Nzqx3syv:b-lcfhfx) ([raw file](https://github.com/cockroachdb/cockroach/blob/d22da89bb21eb4d11c10d88782e9bd827bda2b87/pkg/cmd/roachtest/tests/failover.go#L0)):*
<details><summary><i>Previously, erikgrinaker (Erik Grinaker) wrote…</i></summary><blockquote>

*(Reviewable was [unable](https://github.com/Reviewable/Reviewable/issues/14) to map [this GitHub inline comment thread](https://github.com/cockroachdb/cockroach/pull/112123#discussion_r1354580056) to the right spot — sorry!)*

Is it ok to not disable this any more? We're going to get stuck if it tries to rebalance something after the restart, no?
</blockquote></details>

I ran it a couple of times and it worked. I think the reason it works is because all the ranges are restricted to either r1-3 or r4-6 by configs, so there is no way to move them around. Let me run 10 more runs of it to make sure they all pass. How would I know if something gets "stuck". The rebalance needs to be disabled otherwise after restart, the first node to start gets all the leases, and they don't get moved around, so we never get leader/leaseholder splits.

**Review by erikgrinaker — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/112123)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andrewbaptist, @DarrylWong, and @herkolategan)

___
*[`pkg/cmd/roachtest/tests/failover.go` line 0 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/112123#gh-1354580056:-NgU88SUEfioMBiz-pZU:b6ly8kg) ([raw file](https://github.com/cockroachdb/cockroach/blob/d22da89bb21eb4d11c10d88782e9bd827bda2b87/pkg/cmd/roachtest/tests/failover.go#L0)):*
>How would I know if something gets "stuck".

These would stall, I think:

https://github.com/cockroachdb/cockroach/blob/5eb4a53a0d1e6d2c9e49c4382f931e98bbe0968d/pkg/cmd/roachtest/tests/failover.go#L621-L624

We can also leave it to the nightlies to find out.

**Comment by erikgrinaker** [comment]
PS: thoughts on why this started flaking so much now?

**Comment by andrewbaptist** [comment]
bors r+

I'm going to merge this so that we can figure out the problem later with nightlies. I can't run roachtests this morning and initial testing seems to show that we don't ever do anything in those relocate lines (nothing escaped constraints for me). If anything we should track down why things were previously escaping constraints and fix them.

**Comment by erikgrinaker** [comment]
Yeah, the escaped constraints only happened once in a while. Let's leave it to the nightlies to figure out.

**Pull request merged**